src/main.rs

```diff
@@ -172,16 +173,28 @@ async fn main() -> Result<(), errors::Error> {
 
     let args = Args::parse();
 
+    let credentials_mode = if let Some(aws_role_arn) = &args.aws_role_arn {
```

Comment:
_question_: priority is then made on role based creds, can you add this in the documentation? Or add a check that both are not set in config and triggers an error if so in order to make it clear to end user.

Reply:
> _question_: priority is then made on role based creds, can you add this in the documentation? Or add a check that both are not set in config and triggers an error if so in order to make it clear to end user.

Alright, I'll add this in the PR shortly.

Reply:
I managed to make them mutually exclusive, but also make sure that one is set. Maybe there's a better way to do this.

I also made non required the boolean args.